lib/cdo/rack/request.rb

```diff
@@ -1,5 +1,20 @@
+require 'rack/request'
+require 'ipaddr'
+
 module Rack
+  module CdoExtension
+    TRUSTED_PROXIES = JSON.parse(IO.read(deploy_dir('lib/cdo/trusted_proxies.json')))['ranges'].map do |proxy|
+      IPAddr.new(proxy)
+    end
+
+    def trusted_proxy?(ip)
+      super(ip) || TRUSTED_PROXIES.any?{|proxy| proxy === ip}
+    end
+  end
+
   class Request
+    prepend CdoExtension
```

Comment:
Since `Request` is in the `Rack` module, `CdoExtension` will resolve to `Rack::CdoExtension` before any other namespaces anyway. However it could possibly collide if we define Rack::CdoExtension elsewhere, so I'll move the module to `Rack::Request::CdoExtension` to make it a more localized namespace.